### DISCUSSION TITLE
error response body inconsistency by using `@fastify/error`

### DISCUSSION THREAD

**Discussion by erfanium** [fastify/fastify#4602] [General]
let's say we have this sample code.

```js
import { createError } from '@fastify/error';

const app = fastify();

const MY_CUSTOM_ERROR = createError('MY_CUSTOM_ERROR', 'description', 400);

app.route({
  method: 'POST',
  url: '/',
  schema: {
    body: { type: "object", properties: { foo: { type: "string" } } }
  },
  handler() {
    throw new MY_CUSTOM_ERROR();
  }
})
```

in case of validation error, the response body would be like this:
```json
{
  "statusCode": 400,
  "error": "Bad Request",
  "message": "body must have required property 'foo'"
}
```

but in case of throwing my custom error, the response body would be like this:
```json
{
  "statusCode": 400,
  "code": "MY_CUSTOM_ERROR"
  "error": "Bad Request",
  "message": "description"
}
```
as you can see, `code` key is not available in framework level errors.
this behavior is already confused our front-end developers and i'm kinda agree with them.
what's is your suggestion? I think we could add `code` key on every framework level errors

**Comment by metcoder95** [comment]
Fastify, thanks to #4554 already is handling that in a standardized way across its errors.
The place that it seems I missed when I was listing the errors is:
https://github.com/fastify/fastify/blob/ec173331c76b8cb31baa5d8764cee9f4f9b6421c/lib/validation.js#L135-L139

Which is in fact the validation step, that you're making an example of. I'll create an issue for that
